content/blog/case-weights/index.Rmd

```diff
@@ -0,0 +1,310 @@
+* [ ] `usethis::use_tidy_thanks()`
+-->
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library(doMC)
+registerDoMC(cores = 10)
+library(tidymodels)
+tidymodels_prefer()
+theme_set(theme_bw())
+```
+
+After quite a bit of time, the tidymodels packages have finally enabled the use of case weights. There has been a ton of work and multiple technical hurdles to overcome. The diversity of the types of weights, and how they should be used, is very complex, but I think that we've come up with a solution that is fairly straight-forward for users. 
+
+## What are case weights?
+
+Case weights are non-negative numbers that are used to impact how much each observation influences a model fit. 
+
+There are different types of case weights whose terminology can be very different across problem domains. Here are some examples: 
+
+* **Frequency weights** are integers that denote how many times a particular row of data has been observed. They help compress redundant rows into a smaller format. 
+* **Importance weights** focus on how much each row of the data set should influence the model fit. These can be based on data or arbitrarily set to achieve some goal. 
+* When survey respondents have different probabilities of selection, (inverse) **probability weights** can help remove bias from the results of a data analysis. 
+* If a data point has an associated precision, **analytic weighting** helps a model focus on the data points with less uncertainty (such as in meta-analysis).
+
+There are undoubtedly more types of weights in other domains. 
+
+Generally speaking, case weights are usually non-negative. 
+
+## How are they used in base R?
+
+The best example is for categorical data where a small number of possible categories are observed many times. For example, the `UCBAdmissions` contains "Aggregate data on applicants to graduate school at Berkeley for the six largest departments in 1973 classified by admission and sex."
+
+```{r UCBAdmissions}
+data("UCBAdmissions")
+UCBAdmissions
+```
+
+This is a 3D array, so let's convert to a rectangular data format:
+
+```{r df}
+library(tidymodels)
+
+ucb <- 
+  as_tibble(UCBAdmissions) %>% 
+  mutate(across(where(is.character), ~ as.factor(.)))
+ucb
+```
+
+There are `r nrow(ucb)` possible configurations of the variables but a total of `r sum(ucb$n)` observations. If we want to model the data in this format, using a logistic regression, we could  use
+
+```{r glm}
+glm_fit <-
+  glm(
+    Admit ~ Gender + Dept,
+    data = ucb,
+    weights = n,
+    family = "binomial"
+  )
+glm_fit
+```
+
+
+_This is not quite right though_. There are 12 combinations of `Gender` and `Dept`. How can the model have 23 total degrees of freedom?
+
+If we are treating our data as binomial, the traditional method for fitting this model is to convert the data to a format with columns for the number of events and non-events (per covariate pattern). Let's convert our data into that format:
```

Comment:
I'm used to seeing the term "predictor" (not "covariate") in tidymodels documentation. Is the choice of "covariate" intentional here?

Reply:
I think of "covariate" being more appropriate for an inferential model and "predictor" for predictive models.